UnitTests/AppAuthTV/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,233 @@
+  XCTAssertEqualObjects(authRequest.scope, testScopeString);
+  XCTAssertEqualObjects(authRequest.additionalParameters, testAdditionalParameters);
+  XCTAssertEqualObjects(authRequest.responseType, OIDResponseTypeCode);
+  XCTAssertEqualObjects(authRequest.redirectURL, [[NSURL alloc] init]);
+}
+
+/*! @brief Tests the @c URLRequest method on a request with no scopes or additional parameters
+ */
+- (void)testURLRequestBasicClientAuth {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  NSURLRequest *URLRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects(URLRequest.HTTPMethod, kHTTPPost);
+  XCTAssertEqualObjects([URLRequest valueForHTTPHeaderField:kHTTPContentTypeHeaderKey],
+                        kHTTPContentTypeHeaderValue);
+  XCTAssertEqualObjects(URLRequest.URL.absoluteString, kInitializerTestTVAuthEndpoint);
+
+  NSDictionary<NSString *, NSString *> *bodyParameters =
+      [self bodyParametersFromURLRequest:URLRequest];
+  NSDictionary<NSString *, NSString *> *expectedParameters = @{@"client_id" : kTestClientID};
+
+  XCTAssertEqualObjects(bodyParameters, expectedParameters);
+}
+
+/*! @brief Tests the @c URLRequest method on a request with two scopes and no additional parameters
+ */
+- (void)testURLRequestScopes {
+  NSArray<NSString *> *testScopes = @[ kTestScope, kTestScopeA ];
+  NSString *testScopeString = [OIDScopeUtilities scopesWithArray:testScopes];
+
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:@[ kTestScope, kTestScopeA ]
+                                          additionalParameters:nil];
+
+  NSURLRequest *URLRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects([URLRequest HTTPMethod], kHTTPPost);
+  XCTAssertEqualObjects([URLRequest valueForHTTPHeaderField:kHTTPContentTypeHeaderKey],
+                        kHTTPContentTypeHeaderValue);
+  XCTAssertEqualObjects(URLRequest.URL.absoluteString, kInitializerTestTVAuthEndpoint);
+
+  NSDictionary<NSString *, NSString *> *bodyParameters =
+      [self bodyParametersFromURLRequest:URLRequest];
+  NSDictionary<NSString *, NSString *> *expectedParameters = @{
+    kTestClientIDKey : kTestClientID,
+    kTestScopeKey : [testScopeString
+        stringByAddingPercentEncodingWithAllowedCharacters:[OIDURLQueryComponent
```

Comment:
nit: This looks a bit messy. Can your put the result of stringByAddingPercentEncodingWithAllowedCharacters: into a local variable and use the variable in the dictionary?